### PR TITLE
Update `find` command test cases

### LINKED ISSUE #259: Update `find` command test cases
* Refactor test cases and extract into different methods
* Add and update test cases where needed
* Add comments to explain use of test cases

### PR DESCRIPTION
**Pull request by elijah-ng**
Closes #259

### CODE REVIEW

File: src/test/java/seedu/address/model/person/AddressContainsKeywordsPredicateTest.java [hunk 1]
```diff
@@ -42,18 +42,23 @@ public void equals() {
...
+        // Boundary case: Multiple keywords, all match.
         predicate = new AddressContainsKeywordsPredicate(Arrays.asList("clementi", "ave"));
         assertTrue(predicate.test(new PersonBuilder().withAddress("311, Clementi Ave 2, #02-25").build()));
 
-        // Only one matching keyword
+        // Duplicate keywords, all match.
+        predicate = new AddressContainsKeywordsPredicate(Arrays.asList("clementi", "clementi"));
+        assertTrue(predicate.test(new PersonBuilder().withAddress("311, Clementi Ave 2, #02-25").build()));
+
+        // Multiple keywords, only one matching keyword.
```
Comment (BenedTj) [inline src/test/java/seedu/address/model/person/AddressContainsKeywordsPredicateTest.java]: I believe these can be annotated as being in the same partition where there is more than one keyword. What do you think?

File: src/test/java/seedu/address/model/person/PersonMatchesFilterPredicateTest.java [hunk 2]
```diff
@@ -63,59 +63,72 @@ public void equals() {
...
+        // One name filter. For each filter, keyword match.
+        PersonMatchesFilterPredicate predicate = new PersonMatchesFilterPredicate(Arrays.asList("tan"),
+                Collections.emptyList(), Collections.emptyList());
         assertTrue(predicate.test(new PersonBuilder()
                 .withName("Alice Tan")
-                .withAddress("311, Clementi Ave 2, #02-25").build()));
+                .withAddress("311, Clementi Ave 2, #02-25")
+                .withTags("West").build()));
 
+        // One address filter. For each filter, keyword match.
```
Comment (BenedTj) [inline src/test/java/seedu/address/model/person/PersonMatchesFilterPredicateTest.java]: This could be annotated as a boundary value since it has a filter with one query in it. What do you think?

File: src/test/java/seedu/address/model/person/PersonMatchesFilterPredicateTest.java [hunk 3]
```diff
@@ -405,10 +439,13 @@ public void test_personDoesNotMatchFilters_returnsFalse() {
                 .withName("Alice Tan")
                 .withAddress("311, Clementi Ave 2, #02-25")
                 .withTags("West").build()));
+    }
 
+    @Test
```
[the remark this answers is not in the document]
Comment (elijah-ng) [reply src/test/java/seedu/address/model/person/PersonMatchesFilterPredicateTest.java]: Similar to above.

File: src/test/java/seedu/address/model/person/AddressContainsKeywordsPredicateTest.java (hunk 1 above)
Comment (BenedTj) [reply src/test/java/seedu/address/model/person/AddressContainsKeywordsPredicateTest.java]: I was thinking of categorizing the behavior based on the number of keywords given but this can also serve as a good way of partitioning the test inputs. Therefore, I consider this resolved.
